tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,343 @@
+            new_configs.append(c)
+        elif "RedshiftDataSourceCreator" in c.offline_store_creator:
+            for p in ["local", "aws"]:
+                new_configs.append(replace(c, provider=p))
+        elif "BigQueryDataSourceCreator" in c.offline_store_creator:
+            for p in ["local", "gcp"]:
+                new_configs.append(replace(c, provider=p))
+    return new_configs
+
+
+@contextmanager
+def construct_test_environment(
+    test_repo_config: TestRepoConfig,
+    create_and_apply: bool = False,
+    materialize: bool = False,
+) -> Environment:
+    """
+    This method should take in the parameters from the test repo config and created a feature repo, apply it,
+    and return the constructed feature store object to callers.
+
+    This feature store object can be interacted for the purposes of tests.
+    The user is *not* expected to perform any clean up actions.
+
+    :param test_repo_config: configuration
+    :return: A feature store built using the supplied configuration.
+    """
+    df = create_dataset()
+
+    project = f"test_correctness_{str(uuid.uuid4()).replace('-', '')[:8]}"
+
+    module_name, config_class_name = test_repo_config.offline_store_creator.rsplit(
+        ".", 1
+    )
+
+    offline_creator: DataSourceCreator = importer.get_class_from_type(
+        module_name, config_class_name, "DataSourceCreator"
+    )(project)
+    ds = offline_creator.create_data_source(
+        project, df, field_mapping={"ts_1": "ts", "id": "driver_id"}
+    )
+    offline_store = offline_creator.create_offline_store_config()
+    online_store = test_repo_config.online_store
+
+    with tempfile.TemporaryDirectory() as repo_dir_name:
+        config = RepoConfig(
+            registry=str(Path(repo_dir_name) / "registry.db"),
+            project=project,
+            provider=test_repo_config.provider,
+            offline_store=offline_store,
+            online_store=online_store,
+            repo_path=repo_dir_name,
+        )
+        fs = FeatureStore(config=config)
+        environment = Environment(
+            name=project,
+            test_repo_config=test_repo_config,
+            feature_store=fs,
+            data_source=ds,
+            data_source_creator=offline_creator,
+        )
+
+        fvs = []
+        entities = []
+        try:
+            if create_and_apply:
+                entities.extend([driver(), customer()])
+                fvs.extend(
+                    [
+                        environment.driver_stats_feature_view(),
+                        environment.customer_feature_view(),
+                    ]
+                )
+                fs.apply(fvs + entities)
+
+            if materialize:
+                fs.materialize(environment.start_date, environment.end_date)
+
+            yield environment
+        finally:
+            offline_creator.teardown()
+            fs.teardown()
+
+
+def parametrize_e2e_test(e2e_test):
```

Comment:
I made a similar comment above at `vary`, but it's not clear to me why we need these `parametrize_x` functions. We already have parameterization in pytest, and if it's about dynamic marking it's pretty straightforward

```python
import pytest


@pytest.mark.parametrize(
    ("n", "expected"),
    [
        pytest.param(1, 3, marks=[pytest.mark.local, pytest.mark.integration]),
        pytest.param(1, 3, marks=[pytest.mark.aws, pytest.mark.integration]),
    ],
)
def test_increment(n, expected):
    assert n + 1 == expected
```

I just want to make sure it's 100% obvious by looking at a test file what is happening. The more we abstract away the test cases from the test code the slower it is to understand and develop tests.